src/librustdoc/html/sources.rs

```diff
@@ -86,17 +86,16 @@ impl SourceCollector<'_, '_> {
             return Ok(());
         }
 
-        let mut contents = match fs::read_to_string(&p) {
+        let contents = match fs::read_to_string(&p) {
             Ok(contents) => contents,
             Err(e) => {
                 return Err(Error::new(e, &p));
             }
         };
 
         // Remove the utf-8 BOM if any
-        if contents.starts_with('\u{feff}') {
-            contents.drain(..3);
-        }
+        let contents =
+            if contents.starts_with('\u{feff}') { &contents[3..] } else { &contents[..] };
```

Comment:
Wouldn't `&contents` be enough in the `else`?

Reply:
I believe that's a `String`

Reply:
Also, there was a bit of back-and-forth, the "previous" code was snuck in a few months ago. This "new" version is exactly the same as before that change.

Reply:
Oh I misunderstood your comment. Fixed.